evm/src/executor/inspector/cheatcodes/env.rs

```diff
@@ -104,6 +104,42 @@ fn accesses(state: &mut Cheatcodes, address: Address) -> Bytes {
     }
 }
 
+#[derive(Clone, Debug, Default)]
+pub struct RecordedLogs {
+    pub entries: Vec<RawLog>,
+}
+
+fn start_record_logs(state: &mut Cheatcodes) {
+    state.recorded_logs = Some(Default::default());
+}
+
+fn get_recorded_logs(state: &mut Cheatcodes) -> Bytes {
```

Comment:
Imo this should clear the `state.recorded_logs` vec (see reasoning below)

Reply:
Will copy `state.recorded_logs` and then call `start_record_logs()` to reset them.

Reply:
You can also use `std::mem::take` instead